libs/core/src/styles/createTheme.tsx

```diff
@@ -384,9 +411,169 @@ const defaultOptions: ThemeOptions = {
+                  }
+                : {
+                    boxShadow: `0 0 8px 0 ${alpha(getColorFromThemeWithColorProps(theme, ownerState), 0.3)}`,
+                  }),
+            },
+          },
+
+          ...(ownerState.warning && {
+            background: theme.palette.warning[50],
+
+            [`& .${outlinedInputClasses.notchedOutline}`]: {
+              borderColor: theme.palette.warning.main,
+            },
+          }),
+
+          [`&.${outlinedInputClasses.error}`]: {
+            background: theme.palette.error[50],
+
+            [`& .${outlinedInputClasses.notchedOutline}`]: {
+              borderColor: theme.palette.error.main,
+            },
+          },
+
+          [`&.${outlinedInputClasses.disabled}`]: {
+            background: theme.palette.grey[50],
+
+            [`& .${outlinedInputClasses.notchedOutline}`]: {
+              borderWidth: '1px',
+              borderColor: theme.palette.grey[200],
+            },
+          },
+        }),
+
+        notchedOutline: ({ theme }) => ({
+          borderWidth: '1px',
+          borderColor: theme.palette.warmGrey[300],
+        }),
+
+        adornedStart: ({ theme, ownerState }) => ({
+          [`& .${svgIconClasses.root}`]: {
+            ...(ownerState.warning && {
+              color: theme.palette.warning.dark,
+            }),
+
+            ...(ownerState.error && {
+              color: theme.palette.error.main,
+            }),
+          },
+        }),
+
+        adornedEnd: ({ theme, ownerState }) => ({
+          [`& .${svgIconClasses.root}`]: {
+            ...(ownerState.warning && {
+              color: theme.palette.warning.dark,
+            }),
+
+            ...(ownerState.error && {
+              color: theme.palette.error.main,
+            }),
+          },
+        }),
+
+        input: ({ theme, ownerState }) => ({
+          height: '1.5rem',
+          paddingLeft: theme.spacing(1),
+          paddingRight: theme.spacing(1),
+
+          ...(ownerState.radius === 24 && {
+            ...(!ownerState.startAdornment && { paddingLeft: theme.spacing(2) }),
+            ...(!ownerState.endAdornment && { paddingRight: theme.spacing(2) }),
+          }),
+
+          fontSize: ownerState.size === 'small' ? '0.75rem' : '0.875rem',
+
+          '&::placeholder': {
+            fontSize: ownerState.size === 'small' ? '12px' : '14px',
+            color: theme.palette.warmGrey[ownerState.disabled ? 200 : 300],
+          },
+
+          //spacing
+          ...(ownerState.size === 'small'
+            ? {
+                paddingTop: theme.spacing(0.5),
+                paddingBottom: theme.spacing(0.5),
+              }
+            : ownerState.size === 'medium'
+            ? {
+                paddingTop: theme.spacing(1),
+                paddingBottom: theme.spacing(1),
+              }
+            : {
+                paddingTop: theme.spacing(1.5),
+                paddingBottom: theme.spacing(1.5),
+              }),
```

Comment:
```suggestion
          paddingTop: theme.spacing(textFieldSpacing[ownerState.size]),
          paddingBottom: theme.spacing(textFieldSpacing[ownerState.size]),
```
textFieldSpacing or any naming that you like.